### ISSUE TITLE
whenever appears to be scheduling everything on all of the servers

### ISSUE THREAD

**Issue by jmartin-sul** [sul-dlss/preservation_catalog#1117]
for example, the queue population and cache cleaner jobs are in the prod 01 crontab.  but our intent is to have queue population only happen on prod 02, and to only have cache cleanup happen on prod 04.

maybe we're not defining our cap roles and our whenever_roles correctly?

debugging suggestion from @atz to narrow things down:
* manually clear the crontab on all 4 prod boxes
* re-deploy the app
* see what the new crontab is

if the new crontab is correct for each machine, that indicates a problem with whenever updating the cron jobs it manages (which is a known issue when redefining cron entries managed by whenever).

if the crontab is incorrect again (e.g. all the machines have all the scheduled jobs listed), that'd indicate that we're either configuring things incorrectly, or that there's a but in whenever.

**Comment by jmartin-sul** [comment]
it seems like the big problem here is noisy reporting from cron daemon:  resque lock's queue de-duplication prevents jobs from getting enqueued too many times, and the cache cleaner ends up trying to delete things that another instance of the cache cleaner already got to first (noisy, but otherwise harmless).

**Comment by atz** [comment]
Deploying to production includes the lines:
```
00:43 whenever:update_crontab
      01 bundle exec whenever --update-crontab preservation_catalog_prod --set environment=production --roles=app,resque,cache_cleaner 
      02 bundle exec whenever --update-crontab preservation_catalog_prod --set environment=production --roles=app,resque,queue_populator 
      01 [write] crontab file updated
    ✔ 01 pres@preservation-catalog-prod-04.stanford.edu 0.563s
      02 [write] crontab file updated
    ✔ 02 pres@preservation-catalog-prod-02.stanford.edu 0.580s
```
It seems to not even have tried to update 1 and 3, which would be correct since neither of those machines has a role w/ any jobs from `schedule.rb` attached.  That is also probably why prod 01 and 03 crontabs weren't getting cleared out by deployment: there were no lines to insert, so no operation was performed.

I tried to do `bundle exec cap stage whenever:clear_crontab`, but that failed:
```
cap aborted!
SSHKit::Runner::ExecuteError: Exception while executing as pres@preservation-catalog-stage-01.stanford.edu: bundle exit status: 10
bundle stdout: Could not locate Gemfile or .bundle/ directory
bundle stderr: Nothing written
```
Doesn't seem to know what the current directory of the app is.  I did not attempt the same w/ production because we have manual bespoke lines in various crontabs that would be nuked.

I removed the whenever lines from prod 01 crontab, i commented out the manually (duplicative!) lines from prod 03.  Deployment did **not** replace them (as noted above).
So I conclude we were getting spam error messages because lines were manually added to too many crontabs and then duplicated by the whenever jobs.  predictable logical collisions.  Now I see:
- prod 02 has 4x `queue_populator` cron tasks
- prod 04 has 2x `cache_cleaner` tasks

That looks correct.

**TLDR**: don’t do crontab stuff manually and then forget about it when it gets put in whenever.  also don’t do the same find/delete manually on multiple systems.  I think we’re good now.
